Test Failure (20180420-0319): ConcurrentDBTest.testGlobalTranSuspendAndResume Access denied ("java.util.PropertyPermission" "user.dir" "read")

Test fails when Java 2 security is enabled because it uses Derby without granting permissions to it.

Test Failure (20180420-0319): com.ibm.ws.concurrent.fat.db.ConcurrentDBTest.testGlobalTranSuspendAndResume

junit.framework.AssertionFailedError: 2018-04-20-07:04:58:550 The response did not contain "[SUCCESS]".  Full output is:"
ERROR: Caught exception attempting to call test method testGlobalTranSuspendAndResume on servlet concurrent.fat.db.web.ConcurrentDBTestServlet
java.sql.SQLException: Java exception: 'Access denied ("java.util.PropertyPermission" "user.dir" "read"): java.security.AccessControlException'. DSRA0010E: SQL State = XJ001, Error Code = 0
	at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.Util.seeNextException(Unknown Source)
	at org.apache.derby.impl.jdbc.Util.javaException(Unknown Source)
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedConnection.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedConnection.<init>(Unknown Source)
	at org.apache.derby.jdbc.InternalDriver.getNewEmbedConnection(Unknown Source)
	at org.apache.derby.jdbc.InternalDriver.connect(Unknown Source)
	at org.apache.derby.jdbc.BasicEmbeddedDataSource40.getConnection(Unknown Source)
	at org.apache.derby.jdbc.BasicEmbeddedDataSource40.getConnection(Unknown Source)
	at org.apache.derby.jdbc.BasicEmbeddedDataSource40.setupResourceAdapter(Unknown Source)
	at org.apache.derby.jdbc.EmbeddedXADataSource.getXAConnection(Unknown Source)
	at com.ibm.ws.rsadapter.impl.DatabaseHelper$1.run(DatabaseHelper.java:946)
	at java.security.AccessController.doPrivileged(AccessController.java:650)
	at com.ibm.ws.rsadapter.impl.DatabaseHelper.getPooledConnection(DatabaseHelper.java:958)
	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.getConnection(WSManagedConnectionFactoryImpl.java:794)
	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.createManagedConnection(WSManagedConnectionFactoryImpl.java:651)
	at com.ibm.ejs.j2c.FreePool.createManagedConnectionWithMCWrapper(FreePool.java:1354)
	at com.ibm.ejs.j2c.FreePool.createOrWaitForConnection(FreePool.java:1228)
	at com.ibm.ejs.j2c.PoolManager.reserve(PoolManager.java:1434)
	at com.ibm.ejs.j2c.ConnectionManager.allocateMCWrapper(ConnectionManager.java:573)
	at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:309)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:143)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:116)
	at concurrent.fat.db.web.ConcurrentDBTestServlet.testGlobalTranSuspendAndResume(ConcurrentDBTestServlet.java:74)
	at componenttest.app.FATServlet.doGet(FATServlet.java:71)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:743)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1154)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:4962)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:996)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1011)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:414)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:373)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:530)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:464)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:329)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:300)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:165)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:74)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:501)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:571)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:926)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1015)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1153)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:785)
Caused by: ERROR XJ001: Java exception: 'Access denied ("java.util.PropertyPermission" "user.dir" "read"): java.security.AccessControlException'.
	at org.apache.derby.iapi.error.StandardException.newException(Unknown Source)
	at org.apache.derby.impl.jdbc.SQLExceptionFactory.wrapArgsForTransportAcrossDRDA(Unknown Source)
Caused by: java.security.AccessControlException: Access denied ("java.util.PropertyPermission" "user.dir" "read")
	at java.security.AccessController.throwACE(AccessController.java:157)
	at java.security.AccessController.checkPermissionHelper(AccessController.java:217)
	at java.security.AccessController.checkPermission(AccessController.java:349)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:562)
	at java.lang.SecurityManager.checkPropertyAccess(SecurityManager.java:1307)
	at java.lang.System.getProperty(System.java:443)
	at java.lang.System.getProperty(System.java:427)
	at java.io.UnixFileSystem.resolve(UnixFileSystem.java:144)
	at java.io.File.getCanonicalPath(File.java:629)
	at org.apache.derby.impl.io.VFMemoryStorageFactory.init(Unknown Source)
	at org.apache.derby.impl.services.monitor.StorageFactoryService.privGetStorageFactoryInstance(Unknown Source)
	at org.apache.derby.impl.services.monitor.StorageFactoryService.access$400(Unknown Source)
	at org.apache.derby.impl.services.monitor.StorageFactoryService$12.run(Unknown Source)
	at org.apache.derby.impl.services.monitor.StorageFactoryService$12.run(Unknown Source)
	at java.security.AccessController.doPrivileged(AccessController.java:650)
	at org.apache.derby.impl.services.monitor.StorageFactoryService.getCanonicalServiceName(Unknown Source)
	at org.apache.derby.impl.services.monitor.BaseMonitor.findProviderAndStartService(Unknown Source)
	at org.apache.derby.impl.services.monitor.BaseMonitor.startPersistentService(Unknown Source)
	at org.apache.derby.iapi.services.monitor.Monitor.startPersistentService(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedConnection.bootDatabase(Unknown Source)
". expected:<0> but was:<1>
	at componenttest.topology.utils.HttpUtils.findStringInHttpConnection(HttpUtils.java:537)
	at componenttest.topology.utils.HttpUtils.findStringInHttpConnection(HttpUtils.java:495)
	at componenttest.topology.utils.HttpUtils.findStringInUrl(HttpUtils.java:472)
	at componenttest.topology.utils.HttpUtils.findStringInReadyUrl(HttpUtils.java:444)
	at componenttest.topology.utils.HttpUtils.findStringInReadyUrl(HttpUtils.java:414)
	at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:81)
	at componenttest.custom.junit.runner.SyntheticServletTest.invokeExplosively(SyntheticServletTest.java:40)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:193)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:315)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:167)

--- a/dev/com.ibm.ws.concurrent_fat_db/publish/servers/concurrent_fat_db/server.xml
+++ b/dev/com.ibm.ws.concurrent_fat_db/publish/servers/concurrent_fat_db/server.xml
@@ -27,5 +27,7 @@
 	<library id="DerbyLib">
 	    <fileset dir="${shared.resource.dir}/derby" includes="derby.jar"/>
 	</library>
-    
+
+    <!-- permissions for Derby -->
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="java.security.AllPermission"/>
 </server>

